File: domain-modules/bulo-flows-module/bulo-flows-data-module/src/main/java/com/maukaim/bulo/flows/data/models/flow/Flow.java
package com.maukaim.bulo.flows.data.models.flow;

import com.maukaim.bulo.commons.models.FlowInterface;

import java.util.Set;

public class Flow implements FlowInterface {
    private final String flowId;
    private final Set<OwnerKey> ownerKeys;
    private final Set<FlowStage> flowStages;
    private final Set<FailureAlternativeRoute> failureAlternativeRoutes;
    private final boolean allowParallelRun;

    public Flow(String flowId,
                Set<OwnerKey> ownerKeys,
                Set<FlowStage> flowStages,
                Set<FailureAlternativeRoute> failureAlternativeRoutes,
                boolean allowParallelRun) {
        this.flowId = flowId;
        this.failureAlternativeRoutes = failureAlternativeRoutes;
        this.allowParallelRun = allowParallelRun;
        this.ownerKeys = ownerKeys;
        this.flowStages = flowStages;
    }

    public Set<OwnerKey> getOwnerKeys() {
        return ownerKeys;
    }

    public String getFlowId() {
        return flowId;
    }

    public Set<FlowStage> getFlowStages() {
        return flowStages;
    }

    public boolean isParallelRunAllowed() {
        return allowParallelRun;
    }

    public Set<FailureAlternativeRoute> getFailureAlternativeRoutes() {
        return failureAlternativeRoutes;
    }

    @Override
    public String toString() {
        return "Flow{" +
                "flowId='" + flowId + '\'' +
                ", ownerKeys=" + ownerKeys +
                ", flowStages=" + flowStages +
                ", failureAlternativeRoutes=" + failureAlternativeRoutes +
                ", allowParallelRun=" + allowParallelRun +
                '}';
    }
}

File: front-end/lib/domains/flows/model/flow.dart
import 'flow_stage.dart';
import 'owner_keys.dart';

class Flow {
  final String flowId;
  final List<OwnerKeys> ownerKeys;
  final List<FlowStage> flowStages;
  final bool parallelRunAllowed;

  Flow(
      {required this.flowId, required this.ownerKeys, required this.flowStages, required this.parallelRunAllowed});

  factory Flow.fromJson(Map<String, dynamic> json) {
    return Flow(
        flowId: json['flowId'],
        ownerKeys: [for (var x in json['ownerKeys']) OwnerKeys.fromJson(x)],
        flowStages: [for (var x in json['flowStages']) FlowStage.fromJson(x)],
        parallelRunAllowed: json['parallelRunAllowed']
    );
  }
}
